### PR TITLE
Changes to RequireNamedArgumentsAnalyzer

### PR DESCRIPTION
**Pull request by jbheard**
- Fixed bug with params
- Added/fixed some test cases
- Changed literal param severity from error to info

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs [hunk 1]
```diff
@@ -1,223 +1,247 @@
...
+				// which case it is "unnamed", so ignore it.
+				if( param.Name == "" ) {
+					continue;
+				}
+
+				// C# allows us to create variables with the same names as reserved keywords,
+				// as long as we prefix with @ (e.g. @int is a valid identifier)
+				// So any parameters which are reserved must have the @ prefix
+				string paramName;
+				SyntaxKind paramNameKind = SyntaxFacts.GetKeywordKind( param.Name );
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs]: When there is no match this helper returns `SyntaxKind.None`: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxKindFacts.cs,960

Could we just check for that, and if so add `@`?

File: src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs [hunk 2]
```diff
@@ -1,223 +1,247 @@
...
+				if( SyntaxFacts.GetReservedKeywordKinds().Any( reservedKind => reservedKind == paramNameKind ) ) {
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs]: I couldn't find anything on `ISymbol`/`IParameterSymbol` that indicates that the name is a verbatim name. Looking at where that detection happens the data might still be there... check `MetadataName`? etc., we can investigate on Monday.

**Comment by j3parker** [comment]
Let's break this up into smaller PRs, there are multiple unrelated things happening here. We can walk through that on Monday!

**Comment by jbheard** [comment]
This has been split into multiple smaller PRs.

**Pull request closed without merging**